kroxylicious-filters/kroxylicious-record-validation/src/main/java/io/kroxylicious/proxy/filter/validation/validators/bytebuf/KarapaceSchemaBytebufValidator.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter.validation.validators.bytebuf;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.apache.kafka.common.record.Record;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.proxy.filter.validation.validators.Result;
+
+public class KarapaceSchemaBytebufValidator implements BytebufValidator {
+    private final HttpClient httpClient;
+    private final String registryUrl;
+    private final int schemaId;
+    private final byte magicByte = 0x0;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @SuppressWarnings({ "checkstyle:RegexpSinglelineJava" })
+    public KarapaceSchemaBytebufValidator(Map<String, Object> config, int schemaId) {
+        this.registryUrl = config.get("karapace.registry.url").toString();
+        this.schemaId = schemaId;
+        this.httpClient = HttpClient.newHttpClient();
+    }
+
+    @Override
+    public CompletionStage<Result> validate(ByteBuffer buffer, Record record, boolean isKey) {
+        try {
+            // If the record includes a schema id, validate that it is consistent with the expected schemaId
+            Optional<Integer> extractedSchemaId = extractSchemaIdFromRecord(buffer, record, isKey);
+            if (extractedSchemaId.isEmpty()) {
+                return CompletableFuture
+                        .completedStage(new Result(false, "No schema id in record (%d), expecting %d".formatted(extractedSchemaId.get(), schemaId)));
+            }
+            else if (extractedSchemaId.filter(e -> e != schemaId).isPresent()) {
+                return CompletableFuture
+                        .completedStage(new Result(false, "Unexpected schema id in record (%d), expecting %d".formatted(extractedSchemaId.get(), schemaId)));
+            }
+            return CompletableFuture.completedStage(new Result(true, "Validated successfully"));
+        }
```

Comment:
Unless I am missing something, you are not yet actually validating the key/value actually meet the schema.